pkg/certificate/service.go

```diff
@@ -255,13 +246,14 @@ func New(ctx context.Context, idCfg *config.IdentityConfig) (daemon.Daemon, erro
 			} else {
 				identity = forceInitIdentity
 				keyPEM = forceInitKeyPEM
-				err = idCfg.Reloader.UpdateCertificate([]byte(identity.X509CertificatePEM), keyPEM)
-				if err != nil {
-					log.Errorf("Failed to reload x509 certificate from identity provider: %s", err.Error())
-				}
 			}
 		}
 
+		errUpdate := idCfg.Reloader.UpdateCertificate([]byte(identity.X509CertificatePEM), keyPEM)
+		if errUpdate != nil {
+			log.Errorf("Failed to update x509 certificate into certificate reloader: %s", err.Error())
+		}
```

Comment:
You've set a wrong err.Error() for log
```suggestion
		errUpdate := idCfg.Reloader.UpdateCertificate([]byte(identity.X509CertificatePEM), keyPEM)
		if errUpdate != nil {
			log.Errorf("Failed to update x509 certificate into certificate reloader: %s", errUpdate.Error())
		}
```

Reply:
OR

```suggestion
		if err := idCfg.Reloader.UpdateCertificate([]byte(identity.X509CertificatePEM), keyPEM); err != nil {
			log.Errorf("Failed to update x509 certificate into certificate reloader: %s", err.Error())
		}
```

Reply:
```suggestion
		if errUpdate := idCfg.Reloader.UpdateCertificate([]byte(identity.X509CertificatePEM), keyPEM); errUpdate != nil {
			log.Errorf("Failed to update x509 certificate into certificate reloader: %s", errUpdate.Error())
		}
```